sylvia-derive/src/message.rs

```diff
@@ -1290,32 +1290,7 @@ impl<'a> GlueMessage<'a> {
 
         let variants_cnt = messages_call.len();
 
-        let dispatch_arms = interfaces.interfaces().iter().map(|interface| {
-            let ContractMessageAttr {
-                variant,
-                customs,
-                ..
-            } = interface;
-
-            let ctx = match (msg_ty, customs.has_query) {
-                (MsgType::Exec, true )=> quote! {
-                    ( ctx.0.into_empty(), ctx.1, ctx.2)
-                },
-                (MsgType::Query, true )=> quote! {
-                    ( ctx.0.into_empty(), ctx.1)
-                },
-                _=> quote! { ctx },
-            };
-
-            match (msg_ty, customs.has_msg) {
-                (MsgType::Exec, true) => quote! {
-                    #contract_enum_name:: #variant(msg) => #sylvia ::into_response::IntoResponse::into_response(msg.dispatch(contract, Into::into( #ctx ))?)
-                },
-                _ => quote! {
-                    #contract_enum_name :: #variant(msg) => msg.dispatch(contract, Into::into( #ctx ))
-                },
-            }
-        });
+        let dispatch_arms = interfaces.emit_dispatch_arms(msg_ty);
```

Comment:
Ideally, such refactoring should be one in separate PR. It could be accepted immediately, and this one would be simpler.

Reply:
you are right